src/viam/rpc/dial.py

```diff
@@ -160,8 +162,11 @@ class _Runtime:
     _lib: ctypes.CDLL
     _ptr: ctypes.c_void_p
     _semaphore: PointerCounter = PointerCounter()
+    _lock: threading.Lock = threading.Lock()
 
     def __new__(cls):
+        cls._lock.acquire()
+        cls._semaphore.increment()
```

Comment:
Can we increment the semaphore right before releasing the lock? Just in case there are any errors that pop up in between incrementing and returning, we don't want to have the semaphore to have a falsely-inflated count

Reply:
If _shared has been initialized, then the if doesn't execute and this is already right before the release.

So if an error happens, it's during the initialization of _shared, so that leaves us with a malformed _shared object, which poisons any future use or future invocations, so the semaphore count is the least of our problems.

But I think we want to increment no matter what. Here's why:

In the case that the above happens and the program manages to survive- if the semaphore isn't incremented and something else comes along and cleans up the memory, then the broken instantiation will cause a panic if it's called. If its semaphore is incremented and its ability to connect is broken, then calling it will merely error, not panic.

Reply:
Fair point. I also just reviewed the code again, and with us calling `release` if dial fails to return a socket *should* cover all cases where the program doesn't just crash